src/library/scala/collection/Map.scala

```diff
@@ -327,6 +327,8 @@ object MapOps {
 @SerialVersionUID(3L)
 object Map extends MapFactory.Delegate[Map](immutable.Map) {
   implicit def toLazyZipOps[K, V, CC[X, Y] <: Iterable[(X, Y)]](that: CC[K, V]): LazyZipOps[(K, V), CC[K, V]] = new LazyZipOps(that)
+
+  private val DefaultSentinel: AnyRef = new AnyRef
```

Comment:
I've added `ScalaRunTime.pfMarker` in https://github.com/scala/scala/pull/6799 so we can use a single sentinel value for all implementations.

Reply:
Sound like a good idea. We'll need to merge one PR or the other before we tie them together.